### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by piero-vic**
# Add tests
- Add test for add function
- Add test for remove function

### CODE REVIEW

File: test/app.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { add, remove } from '../src/crud';
+
+describe('Tests for todo list app', () => {
```
Comment (rloterh) [inline test/app.test.js:3]: - [ ] Other _pure functions_ in your code are **not tested** aside _add item_ and _delete item_. Kindly ensure to test **all** pure functions such as the _edit_, _complete: true/false_, _clear_ all completed_ and other created pure functions.

File: test/mock/todoList.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export default class ToDo {
```
Comment (rloterh) [inline test/mock/todoList.js:1]: - [ ] Local storage is not included in your test mock functions. Kindly ensure that `local storage, `DOM` or external APIs uses test functions defined as a mock.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Piero & Jim, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Test cases are passings
- Test covers **add item** and **delete item** functions
- Tests are grouped using the `description()` method


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/app.test.js (hunk 1 above)
Comment (piero-vic) [reply test/app.test.js:3]: Hi. According to the instuctions [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md). We should only implement tests for the add and remove functionality. the other tests are going to be implemented on the [next activity](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_2.md)

File: test/mock/todoList.js (hunk 2 above)
Comment (piero-vic) [reply test/mock/todoList.js:1]: We are using an npm package to handele the local storage mock. It's declared here.
https://github.com/piero-vic/to-do-list/blob/6cb2a47801100ceb6689b90a6015918785db986b/package.json#L38
And the configuration is here.
https://github.com/piero-vic/to-do-list/blob/6cb2a47801100ceb6689b90a6015918785db986b/package.json#L10-L11
Thanks for the feedback.

File: test/app.test.js (hunk 1 above)
Comment (rloterh) [reply test/app.test.js:3]: Hi @piero-vic 
Here is a screenshot of this project requirement (Kindly look at it carefully):
image

File: test/mock/todoList.js (hunk 2 above)
Comment (rloterh) [reply test/mock/todoList.js:1]: Hi @piero-vic 
You are very right on this aspect. I will leave a note for the same!

**Comment by rloterh** [comment]
**Hi Fellow Code Reviewer:**

Kindly note, mock function for local storage was implemented using dependencies, as indicated above. I apologize for missing this important step.

File: test/app.test.js (hunk 1 above)
Comment (piero-vic) [reply test/app.test.js:3]: Hi. Thank you for the response. I appreciate it. I'm having concerns about what are we going to do in the next activity if we implement all the tests in this pull request. This are the requirements for the [next activity](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_2.md).
project-requirements
Thanks you again. I look forward to your reply.

**Comment by piero-vic** [comment]
Hi. Please take into consideration that we are implementing tests just for the add and remove funcitonality. Since the other funtionality it's goint to be tested in the [next activity](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_2.md).
project-requirements-1

**Review by dasileker — APPROVED** [review]
## **Hi @piero-vic,** 

### **Optional**

- As The TSE said the pure function can be added, and also you are correct but you can add more test suits even if you add those.

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**